tests/integration/netdev.bats

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env bats
+
+load helpers
+
+function setup() {
+	requires root
+	setup_busybox
+	# create a dummy interface to move to the container
+	ip link add dummy0 type dummy
+}
+
+function teardown() {
+	ip link del dev dummy0
+	teardown_bundle
+}
+
+@test "move network device to container network namespace" {
+	update_config ' .linux.netDevices |= {"dummy0": {} }
+								| .process.args |= ["ip", "address", "show", "dev", "dummy0"]'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+}
+
+@test "move network device to container network namespace and restore it back" {
+	# create a temporary name for the test network namespace
+	tmp=$(mktemp)
+	rm -f "$tmp"
+	ns_name=$(basename "$tmp")
+	# create network namespace
+	ip netns add "$ns_name"
+	ns_path=$(ip netns add "$ns_name" 2>&1 | sed -e 's/.*"\(.*\)".*/\1/')
+
+	# tell runc which network namespace to use
+	update_config '(.. | select(.type? == "network")) .path |= "'"$ns_path"'"'
+
+	update_config ' .linux.netDevices |= {"dummy0": {} }'
+
+	runc run -d --console-socket "$CONSOLE_SOCKET" test_busybox
+	[ "$status" -eq 0 ]
+
+	# the network namespace owner controls the lifecycle of the interface
+	ip netns exec "$ns_name" ip link set dev dummy0 netns 1
+	[ "$status" -eq 0 ]
+
+	runc delete --force test_busybox
+
+	# verify the interface is back in the root network namespace
+	ip address show dev dummy0
+	[ "$status" -eq 0 ]
```

Comment:
I'm definitely missing something here. Who is deleting the netns? runc delete does that? That seems surprising for me, but I've never dealt with the delete code. If we specified an externally managed netns, why is runc deleting it?

Reply:
the namespace is leaking, there has to be something managing this namespace lifecycle, if you see the other tests I have a `ip netns del "$ns_name"` , it was just missing here

Reply:
moved the network namespace creation and deleting logic to the `setup()` and `teardown()` hooks

Reply:
see the command in line52 , is bringing the interface back to the root namespace manually

Reply:
Ohhh, thanks!